runtime/lua/vim/diagnostic.lua

```diff
@@ -1809,22 +1885,8 @@ end
 --- @param namespace integer
 --- @param bufnr integer
 local function render_virtual_lines_at_current_line(diagnostics, namespace, bufnr)
-  local lnum = api.nvim_win_get_cursor(0)[1] - 1
-  local cursor_diagnostics = {}
-
-  if diagnostics[lnum] ~= nil then
-    cursor_diagnostics = diagnostics[lnum]
-  else
-    for _, line_diags in pairs(diagnostics) do
-      for _, diag in ipairs(line_diags) do
-        if diag.end_lnum and lnum >= diag.lnum and lnum <= diag.end_lnum then
-          table.insert(cursor_diagnostics, diag)
-        end
-      end
-    end
-  end
-
-  render_virtual_lines(namespace, bufnr, cursor_diagnostics)
+  diagnostics = diagnostics_at_cursor(diagnostics)
+  render_virtual_lines(namespace, bufnr, diagnostics)
 end
```

Comment:
Can we remove this function completely and use

```lua
render_virtual_lines(namespace, bufnr, diagnostics_at_cursor(diagnostics))
```

instead?

Reply:
Hmm given that we use this function in 2 places and the name makes the code clearer, I prefer keeping it.

Reply:
It also aligns with how we have a `render_virtual_text_at_current_line`.

The name is a bit long though, I admit. I'm open to renaming it to something less verbose.

Reply:
> Hmm given that we use this function in 2 places and the name makes the code clearer, I prefer keeping it.

I don't actually think it makes the code any clearer. Consider:

```lua
render_virtual_lines(namespace, bufnr, diagnostics_at_cursor(diagnostics))
```

vs

```lua
render_virtual_lines_at_current_line(diagnostics, namespace, bufnr)
```

If anything, I'd argue the extra function makes it less clear because there are two different functions with similar names that do almost the same thing. Having fewer functions that compose well is better than many functions with specific uses.

Reply:
Okay. I don't feel strongly about this so I'll remove the `*_at_current_line` functions if it bothers you.